src/main/java/com/google/cloud/tools/appengine/cloudsdk/serialization/CloudSdkVersion.java

```diff
@@ -52,22 +52,29 @@ public CloudSdkVersion(String version) {
     Preconditions.checkNotNull(version, "Null version");
     Preconditions.checkArgument(!version.isEmpty(), "empty version");
 
-    Matcher matcher = SEMVER_PATTERN.matcher(version);
-    if (!matcher.matches()) {
-      throw new IllegalArgumentException(
-          String.format("Pattern \"%s\" is not a valid CloudSdkVersion.", version));
-    }
-
-    majorVersion = Integer.parseInt(matcher.group("major"));
-    minorVerion = Integer.parseInt(matcher.group("minor"));
-    patchVersion = Integer.parseInt(matcher.group("patch"));
+    if ("HEAD".equals(version.trim())) {
+      majorVersion = Integer.MAX_VALUE;
```

Comment:
Instead of faking a version number we could change the comparable logic to understand that HEAD is later than other versions.

Reply:
It's not a big deal to have such a dedicated logic, but we still have the question of which integer value to return when clients call `getMajorVersion()`, `getMinorVersion()`, and `getPatchVersion()`. And, we should also consider that clients may use these method to determine minimal versions, for example. That is,

```java
if (version.getMajorVersion() > 198
    || version.getMajorVersion() > 197 && version.getMinorVersion() > 3) {
   // minimum version requirement
}
```

Returning large values for HEAD seems not a terrible idea, as the main purpose as I understand is to unblock the internal users from using our tools by making the check for the minimum SDK version requirement pass.

Reply:
That said, another conceptually ideal way is to make `getMajorVersion()` and such return another layer of abstraction on top of `int` to include a special value for the HEAD notion, but considering that this is mainly for internal support (and maybe an SDK manually built from source, which would extremely rare), I don't think it's worth the complexity.